onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -178,20 +184,20 @@ Status Pool<T, PoolType, NHWC>::ComputeInternal(OpKernelContext* context) const
   TensorShapeVector x_dims_cudnn(x_dims.begin(), x_dims.end());
   TensorShapeVector y_dims_cudnn(y_dims);
   if (kernel_shape.size() < 2) {
-    // cudnn only takes 4D or 5D input, so pad dimensions if needed
-    if (NHWC) {
+    // cuDNN only takes 4D or 5D input, so pad dimensions if needed
+    if (Layout == LAYOUT_NHWC) {
       x_dims_cudnn.insert(x_dims_cudnn.begin() + 1, 1);
       y_dims_cudnn.insert(y_dims_cudnn.begin() + 1, 1);
-      kernel_shape.insert(kernel_shape.begin() + 1, 1);
-      strides.insert(strides.begin() + 1, 1);
+      pads.insert(pads.begin(), 0);
+      kernel_shape.insert(kernel_shape.begin(), 1);
+      strides.insert(strides.begin(), 1);
     } else {
-      x_dims_cudnn.push_back(1);
-      y_dims_cudnn.push_back(1);
-      kernel_shape.push_back(1);
-      strides.push_back(1);
+      x_dims_cudnn.insert(x_dims_cudnn.begin() + 2, 1);
+      y_dims_cudnn.insert(y_dims_cudnn.begin() + 2, 1);
+      pads.insert(pads.begin(), 0);
```

Comment:
Is there performance benefit of NC1D vs. NCD1? If yes, I think we can keep NC1D. If not, recommend to pad as NCD1 by default (similar to Conv). Same applies to channel last.

Reply:
I don't have numbers of performance differences. I'll pad at the end as requested